labs/src/main/scala/org/scalalabs/basic/lab03/RecursionPatternMatchingExercise.scala

```diff
@@ -77,7 +105,10 @@ object RecursionPatternMatchingExercise {
    * List(List(1), List('A, 'B, 'C), List('a, 'b)) -> List(List(1, 'A, 'a))
    */
   def zipMultipleWithDifferentSize(in: List[List[_]]): List[List[_]] = {
-    error("fix me")
+    in match {
+      case l if (l.map((a) => a.length)).min == 1 => List(l.map((a) => a.head))
+      case l => List(l.map((a) => a.head)) ++ zipMultipleWithDifferentSize(l.map((a) => a.tail))
+    }
   }
```

Comment:
This seems pretty Scala-y considering its all pattern matching and leverages a lot of the functional programming aspects of Scala. I think it's a little bit hard to read though considering the bulk of the work is being done in one case.